src/app/cluster/details/shared/version-change-dialog/template.html

```diff
@@ -55,6 +55,16 @@
       Upgrade Machine Deployments
     </mat-checkbox>
   </div>
+  <div *ngIf="!isDynamicKubletConfigSupportedInUpgrade"
```

Comment:
Can you add another check here so this only shows up when at least one machine-deployment has `dynamicConfig` enabled?
For example, if there is no machine deployment with `dynamicConfig` enabled then it looks like this:

![screenshot-localhost_8000-2022 06 14-19_05_41](https://user-images.githubusercontent.com/13975988/173639498-cec52fc6-05b6-4e2b-b51c-252b402a7757.png)

Reply:
so you mean like if there is no MD using dynamic kubelet config then sow another warning, like :
`kubernetes version 1.24 and higher do not support dynamic kubelet configuration`
and if there is some MD using dynamic kubelet config then show the current warning 
@Waseem826

Reply:
There are multiple scenarios:

- When `NO` machine deployment exists user tries to add new MD + Kubernetes version 1.24 or > greater should we show “Toggle button” 
(IMO not since all MD will have auto disabled dynamic kubelet configuration

- When `NO`  machine deployment exists user tries to add new MD + Kubernetes version 1.24 < and user tries to enable toggle -> WARNING should be shown. 

- When node/machine are on 1.24 -> Should we hide that toggle?because it doesn’t make sense whether update we need to show warning or not because BE will automatically disables “dynamic kubelet configuration”


cc: @cschieder @embik

Reply:
the warning is not when you create new MD it's on upgrade the kubernetes version to 1.24 in cluster
and if the current version of kubernetes is 1.24 you can't upgrade it for now 
maybe adjust it to bee shown just when upgrde to 1.24 so in the future if the user want to upgrade to (1.24.5 for example) it wouldn't show the warning

Reply:
@ahmadhamzh I think we should hide the warning altogether when there is no machine deployment with dynamic config. i.e. `nodesSupportDynamicKubeletConfig` is empty

Reply:
yes but what if the user want to upgrade to 1.24 and he don't know that he can't use dynamic kubelet config,so we suppose to warn him 
IMO the warning suppose to show when upgrade to 1.24 ,
if any MD using dynamic kubelet config then show the current error 

and if there is no MD using dynamic kubelet config then show the error :
`kubernetes version 1.24 and higher do not support dynamic kubelet configuration`

Reply:
Lets move this conversation to Slack (in sig-api-ui) and ask @cschieder and others about this.